### PR TITLE
Optionally commit sandboxes

### PR DESCRIPTION
**Pull request by gliargovas**
This WIP will solve the backward dependency rollback issue.

### CODE REVIEW

File: test/test_orch.sh [hunk 1]
```diff
@@ -49,9 +49,9 @@ run_test()
     diff -q "$output_dir_orch/" "$output_dir_bash/"
     test_diff_ec=$?
 
-    ## Check if the two exit codes are both success or both error
-    { [ $test_bash_ec -eq 0 ] && [ $test_pash_ec -eq 0 ]; } || { [ $test_bash_ec -ne 0 ] && [ $test_pash_ec -ne 0 ]; }
-    test_ec=$?
+    # ## Check if the two exit codes are both success or both error
```
Comment (angelhof) [inline test/test_orch.sh:52]: Add a TODO that we comment this out until we actually fix exit code handling in the system

File: test/test_orch.sh [hunk 2]
```diff
@@ -118,8 +136,10 @@ if [ "$#" -eq 0 ]; then
     cleanup
     run_test test3
     cleanup
-    ## TODO: Fails at the moment, uncomment when fixed
-    # run_test test4
+    run_test test4
+    cleanup
+    # TODO: Fails at the moment, uncomment when fixed
```
Comment (angelhof) [inline test/test_orch.sh]: Does test 4 or 5 fail?

Comment (gliargovas) [reply test/test_orch.sh]: Test 5 used to fail, now it passes 😃

**Comment by gliargovas** [comment]
@angelhof ready for review, unless we want to also implement W-W dependency optimization here.

**Review by angelhof — APPROVED** [review]
Looks good, we can actually deal wtih the WW optimization here (since you are already commiting sandboxes in order I guess that we just need to not consider W - W conflict a dependency.

**Review by angelhof — APPROVED** [review]
looks great! let's merge

**Pull request merged**